frontend/src/components/note/NoteModal.vue

```diff
@@ -0,0 +1,116 @@
+<script setup lang="ts">
+import { Form } from 'vee-validate';
+// import { ref } from 'vue';
+import { object, string } from 'yup';
+
+import { Dropdown, InputText, TextArea } from '@/components/form';
+import { Button, Dialog } from '@/lib/primevue';
+// import { noteService } from '@/services';
+import { NoteTypes } from '@/utils/constants';
+import { NOTE_TYPES } from '@/utils/enums';
+import { onMounted } from 'vue';
+
+// import type { Ref } from 'vue';
+import type { Note } from '@/types';
+
+// Props
+type Props = {
+  note?: Note;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  note: undefined
+});
+
+// Emits
+const emit = defineEmits(['note:submit']);
+
+// State
+const visible = defineModel<boolean>('visible');
+
+// Default form values
+let initialFormValues: any = {
+  note: props.note?.note,
+  noteType: NOTE_TYPES.GENERAL
+};
+
+// Form validation schema
+const formSchema = object({
+  note: string().required().label('Note'),
+  noteType: string().oneOf(NoteTypes).label('Note type'),
+  title: string().required().max(100, 'Title too long').label('Title')
+});
+
+// Actions
+// @ts-expect-error TS7031
+// resetForm is an automatic binding https://vee-validate.logaretm.com/v4/guide/components/handling-forms/
+function onSubmit(data: any, { resetForm }) {
+  if (props.note) initialFormValues = data;
+  else resetForm();
+  emit('note:submit', data);
```

Comment:
Style: Should have a space between 49 and 50. They could be confused for the same else block